Sources/Extensions/SwiftStdlib/StringProtocolExtensions.swift

```diff
@@ -19,8 +19,28 @@ public extension StringProtocol where Index == String.Index {
     ///     - Parameter options: Options for the comparison.
     /// - Returns: The longest common suffix of the receiver and the given String
     public func commonSuffix<T: StringProtocol>(with aString: T, options: String.CompareOptions = []) -> String {
-        let reversedSuffix = String(reversed()).commonPrefix(with: String(aString.reversed()), options: options)
-        return String(reversedSuffix.reversed())
+        guard !isEmpty && !aString.isEmpty else { return "" }
+
+        var idx = endIndex
+        var strIdx = aString.endIndex
+
+        repeat {
+            formIndex(before: &idx)
+            aString.formIndex(before: &strIdx)
+
+            if String(self[idx]).compare(String(aString[strIdx]), options: options) != .orderedSame { break }
+
+        } while idx > startIndex && strIdx > aString.startIndex
+
+        if idx == startIndex {
+            return String(self)
+        }
+
+        if strIdx == aString.startIndex {
```

Comment:
All these index edge cases and whatnot is why I think it would've been easier to just use my implementation. I guess your imp potentially saves some memory usage and whatnot, but there are still obviously some potential pitfalls.